galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/RestoreCPS.java

```diff
@@ -0,0 +1,294 @@
+        
+        path = Paths.get(filePath);
+        InputStream inputStream = null;
+        try {
+            inputStream = Files.newInputStream(path);
+        } catch (IOException e) {
+            logger.error("Error Creating InputStream Using File: " + path.toUri().toString(), e);
+        }
+        
+        if (inputStream != null) {
+            try {
+                properties.load(inputStream);
+            } catch (IOException e) {
+                logger.error("Error Loading Properties From InputStream", e);
+            }
+        }
+        
+        return properties;
+
+    }
+    
+    /**
+     * <p>Iterates through the properties in the supplied object, restoring them one-by-one to the Configuration Property Store </p>
+     * 
+     * @param properties (instance of object: Properties)
+     * @return 
+     */
+    private void restoreProperties(Properties properties) throws ConfigurationPropertyStoreException {
+        
+        logger.info(String.format("|%-12s|%-17s|%-32s|%-17s|%-17s|", getDashes(12), getDashes(17), getDashes(32), getDashes(17), getDashes(17)));
+        logger.info(String.format("| %-10s | %-15s | %-30s | %-15s | %-15s |", "STATUS", "NAMESPACE", "PROPERTY", "OLD VALUE", "NEW VALUE"));
+        logger.info(String.format("|%-12s|%-17s|%-32s|%-17s|%-17s|", getDashes(12), getDashes(17), getDashes(32), getDashes(17), getDashes(17)));
+        
+        Map<String, String> invalidProperties = new HashMap<String, String>();
+        
+        for (Entry<Object, Object> prop : properties.entrySet()) {
+            if (isValidProperty(prop.getKey().toString())) {
+                restoreProperty(prop);
+            } else {
+                invalidProperties.put(prop.getKey().toString(), prop.getValue().toString());
+            }
+        }
+        logger.info(String.format("|%-12s|%-17s|%-32s|%-17s|%-17s|", getDashes(12), getDashes(17), getDashes(32), getDashes(17), getDashes(17)));
+
+        if (invalidProperties.size() > 0) {
+            logger.info("Could not restore the following due to malformed property name:");
+            for (Entry<String, String> prop : invalidProperties.entrySet()) {
+                logger.info("NAME: " + prop.getKey().toString() + "\tVALUE: " + prop.getValue().toString());
+            }
+        }
+        
+    }
+    
+    /**
+     * <p>Restores invidividual property to CPS</p>
+     * 
+     * @param prop (Map<Object, Object>)
+     * @return
+     * @throws ConfigurationPropertyStoreException
+     */
+    private void restoreProperty(Entry<Object, Object> prop) throws ConfigurationPropertyStoreException {
+        
+        String namespace = getPropertyPrefix(prop.getKey().toString());
+        String property = getPropertySuffix(prop.getKey().toString());
+        
+        String newValue = "";
+        String currentValue = "";
+        
+        String propertyRestoreStatus = "";
+
+        if (isNamespaceRestorePermitted(namespace)){
```

Comment:
Instead of loads of embedded ifs,    do !isNamespaceRestorePermitted,  then return.
SonarQube will not like the complexity.    Also means you can set the status to Invalid name space.

Reply:
Not sure I'm spotting the same simplification as you, especially as we've removed the method `isNamespaceRestorePermitted()`. Are you suggesting we limit the statuses we assign?